pygaggle/data/msmarco.py

```diff
@@ -0,0 +1,143 @@
+import os
+from collections import OrderedDict, defaultdict
+from typing import List, Set, DefaultDict
+import json
+import logging
+from itertools import permutations
+
+from pydantic import BaseModel
+import scipy.special as sp
+import numpy as np
+
+from .relevance import RelevanceExample, MsMarcoPassageLoader
+from pygaggle.model.tokenize import SpacySenticizer
+from pygaggle.rerank.base import Query, Text
+from pygaggle.data.unicode import convert_to_unicode
+
+
+__all__ = ['MsMarcoExample', 'MsMarcoDataset']
+
+
+class MsMarcoExample(BaseModel):
+    qid: str
+    text: str
+    candidates: List[str]
+    relevant_candidates: Set[str]
+
+class MsMarcoDataset(BaseModel):
+    examples: List[MsMarcoExample]
+
+    @classmethod
+    def load_qrels(cls, path: str) -> DefaultDict[str, Set[str]]:
+        qrels = defaultdict(set)
+        with open(path) as f:
+            for i, line in enumerate(f):
+                qid, _, doc_id, relevance = line.rstrip().split('\t')
+                if int(relevance) >= 1:
+                    qrels[qid].add(doc_id)
+                if i % 1000 == 0:
+                    print(f'Loading qrels {i}')
+        return qrels
+
+    @classmethod
+    def load_run(cls, path: str):
+        '''Returns OrderedDict[str, List[str]]'''
+        run = OrderedDict()
+        with open(path) as f:
+            for i, line in enumerate(f):
+                qid, doc_title, rank = line.split('\t')
+                if qid not in run:
+                    run[qid] = []
+                run[qid].append((doc_title, int(rank)))
+                if i % 1000000 == 0:
```

Comment:
maybe replace these debugging lines with `tqdm`?

Reply:
I'll just remove these since they load in a couple of seconds!

Reply:
Cool!